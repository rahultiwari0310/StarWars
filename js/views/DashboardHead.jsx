
import React from 'react';

const DashboardHead = ( props ) => {

    return (
        <h1 className='dashboard__head' >
            <span className='dashboard__head__welcome'>Welcome to Star Wars planets Page.</span>
            <span className='dashboard__head__greeting'> Hello { props.loggedInUser } ! </span>
            <button className='dashboard__head__logout' onClick={ props.handleLogout }>Logout</button>
        </h1>
    );

};

export default DashboardHead;
