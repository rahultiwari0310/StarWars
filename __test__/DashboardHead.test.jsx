import 'raf/polyfill';      // Warning: React depends on requestAnimationFrame. Make sure that you load a polyfill in older browsers. http://fb.me/react-polyfills
import React from 'react';
import ReactDOM from 'react-dom';
import sinon from 'sinon';
import DashboardHead from '../js/views/DashboardHead.jsx';
import Enzyme, { shallow, mount, render } from 'enzyme';
/*
    Fix for Enzyme Internal Error: Enzyme expects an adapter to be configured, but found none. To
    configure an adapter, you should call `Enzyme.configure({ adapter: new Adapter() })`
    before using any of Enzyme's top level APIs, where `Adapter` is the adapter
    corresponding to the library currently being tested.
*/
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('DashboardHead', () => {
    it('renders without error', () => {
        const componentTree = mount(
            <DashboardHead loggenInUser='Luke Skywalker' handleLogout={ ()=>{} }/>
        );
        expect(componentTree.length).toEqual(1);
        expect(componentTree).toMatchSnapshot();
    });

    it('handles logout click correctly', () => {
        const handleLogout = jest.fn(),
            luke = 'Luke Skywalker';
        const componentTree = mount(
            <DashboardHead loggedInUser={ luke } handleLogout={ handleLogout }/>
        );
        componentTree.find( '.dashboard__head__logout' ).simulate( 'click' );
        expect( handleLogout).toHaveBeenCalled();

        //Check if greeting is correctly rendered
        expect( componentTree.find( '.dashboard__head__greeting' ).text()).toEqual( ' Hello ' + luke + ' ! ' );

    });

});