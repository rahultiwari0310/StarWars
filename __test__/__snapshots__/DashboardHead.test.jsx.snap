// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardHead renders without error 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <DashboardHead
    handleLogout={[Function]}
    loggenInUser="Luke Skywalker"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "handleLogout": [Function],
      "loggenInUser": "Luke Skywalker",
    },
    "ref": null,
    "rendered": Object {
      "instance": <h1
        class="dashboard__head"
      >
        <span
          class="dashboard__head__welcome"
        >
          Welcome to Star Wars planets Page.
        </span>
        <span
          class="dashboard__head__greeting"
        >
           Hello 
           ! 
        </span>
        <button
          class="dashboard__head__logout"
        >
          Logout
        </button>
      </h1>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="dashboard__head__welcome"
          >
            Welcome to Star Wars planets Page.
          </span>,
          <span
            className="dashboard__head__greeting"
          >
             Hello 
             ! 
          </span>,
          <button
            className="dashboard__head__logout"
            onClick={[Function]}
          >
            Logout
          </button>,
        ],
        "className": "dashboard__head",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <span
            class="dashboard__head__welcome"
          >
            Welcome to Star Wars planets Page.
          </span>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to Star Wars planets Page.",
            "className": "dashboard__head__welcome",
          },
          "ref": null,
          "rendered": Array [
            "Welcome to Star Wars planets Page.",
          ],
          "type": "span",
        },
        Object {
          "instance": <span
            class="dashboard__head__greeting"
          >
             Hello 
             ! 
          </span>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " Hello ",
              undefined,
              " ! ",
            ],
            "className": "dashboard__head__greeting",
          },
          "ref": null,
          "rendered": Array [
            " Hello ",
            " ! ",
          ],
          "type": "span",
        },
        Object {
          "instance": <button
            class="dashboard__head__logout"
          >
            Logout
          </button>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Logout",
            "className": "dashboard__head__logout",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Logout",
          ],
          "type": "button",
        },
      ],
      "type": "h1",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "handleLogout": [Function],
        "loggenInUser": "Luke Skywalker",
      },
      "ref": null,
      "rendered": Object {
        "instance": <h1
          class="dashboard__head"
        >
          <span
            class="dashboard__head__welcome"
          >
            Welcome to Star Wars planets Page.
          </span>
          <span
            class="dashboard__head__greeting"
          >
             Hello 
             ! 
          </span>
          <button
            class="dashboard__head__logout"
          >
            Logout
          </button>
        </h1>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="dashboard__head__welcome"
            >
              Welcome to Star Wars planets Page.
            </span>,
            <span
              className="dashboard__head__greeting"
            >
               Hello 
               ! 
            </span>,
            <button
              className="dashboard__head__logout"
              onClick={[Function]}
            >
              Logout
            </button>,
          ],
          "className": "dashboard__head",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <span
              class="dashboard__head__welcome"
            >
              Welcome to Star Wars planets Page.
            </span>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to Star Wars planets Page.",
              "className": "dashboard__head__welcome",
            },
            "ref": null,
            "rendered": Array [
              "Welcome to Star Wars planets Page.",
            ],
            "type": "span",
          },
          Object {
            "instance": <span
              class="dashboard__head__greeting"
            >
               Hello 
               ! 
            </span>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " Hello ",
                undefined,
                " ! ",
              ],
              "className": "dashboard__head__greeting",
            },
            "ref": null,
            "rendered": Array [
              " Hello ",
              " ! ",
            ],
            "type": "span",
          },
          Object {
            "instance": <button
              class="dashboard__head__logout"
            >
              Logout
            </button>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Logout",
              "className": "dashboard__head__logout",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "Logout",
            ],
            "type": "button",
          },
        ],
        "type": "h1",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
